src/rating.tsx

```diff
@@ -0,0 +1,349 @@
+import IconFaceNeutralRegular from './generated/mistica-icons/icon-face-neutral-regular';
+import IconFaceHappyFilled from './generated/mistica-icons/icon-face-happy-filled';
+import IconFaceHappyRegular from './generated/mistica-icons/icon-face-happy-regular';
+import IconFaceSuperHappyFilled from './generated/mistica-icons/icon-face-super-happy-filled';
+import IconFaceSuperHappyRegular from './generated/mistica-icons/icon-face-super-happy-regular';
+import {useTheme} from './hooks';
+
+import type {ExclusifyUnion} from './utils/utility-types';
+import type {DataAttributes, IconProps} from './utils/types';
+
+type RatingIconProps = {
+    ActiveIcon: React.FC<IconProps>;
+    InactiveIcon: React.FC<IconProps>;
+    color: string;
+};
+
+const DEFAULT_RATING_SIZE = 32;
+const DEFAULT_INFO_RATING_SIZE = 16;
+const DEFAULT_RATING_ICON_COUNT = 5;
+
+const DEFAULT_RATING_ICON: RatingIconProps = {
+    ActiveIcon: IconStarFilled,
+    InactiveIcon: IconStarRegular,
+    color: vars.colors.controlActivated,
+};
+
+const DEFAULT_INFO_RATING_ICON: RatingIconProps = {
+    ActiveIcon: IconStarFilled,
+    InactiveIcon: IconStarRegular,
+    color: vars.colors.warning,
+};
+
+const DEFAULT_QUALITATIVE_ICONS: Array<RatingIconProps> = [
+    {
+        ActiveIcon: IconFaceSadFilled,
+        InactiveIcon: IconFaceSadRegular,
+        color: vars.colors.errorHigh,
+    },
+    {
+        ActiveIcon: IconFaceSlightlySadFilled,
+        InactiveIcon: IconFaceSlightlySadRegular,
+        color: vars.colors.error,
+    },
+    {
+        ActiveIcon: IconFaceNeutralFilled,
+        InactiveIcon: IconFaceNeutralRegular,
+        color: vars.colors.warning,
+    },
+    {
+        ActiveIcon: IconFaceHappyFilled,
+        InactiveIcon: IconFaceHappyRegular,
+        color: vars.colors.success,
+    },
+    {
+        ActiveIcon: IconFaceSuperHappyFilled,
+        InactiveIcon: IconFaceSuperHappyRegular,
+        color: vars.colors.successHigh,
+    },
+];
+
+interface BaseRatingProps {
+    size?: number;
+    dataAttributes?: DataAttributes;
+    valueLabels?: Array<string>;
+    'aria-label'?: string;
+    'aria-labelledby'?: string;
+}
+
+interface QuantitativeRatingProps extends BaseRatingProps {
+    type?: 'quantitative';
+    icon?: RatingIconProps;
+    count?: number;
+    icons?: never;
+}
+
+interface QualitativeRatingProps extends BaseRatingProps {
+    type: 'qualitative';
+    icons?: Array<RatingIconProps>;
+    icon?: never;
+    count?: never;
+}
+
+type RatingProps = ExclusifyUnion<QualitativeRatingProps | QuantitativeRatingProps> & {
+    value?: number;
+    defaultValue?: number;
+    onChangeValue?: (value: number) => void;
+    disabled?: boolean;
+};
+
+type InfoRatingProps = Omit<QuantitativeRatingProps, 'type' | 'valueLabels'> & {
+    value?: number;
+    withHalfValue?: boolean;
+};
+
+type InternalRatingProps = ExclusifyUnion<RatingProps | InfoRatingProps> & {
+    isInteractive: boolean;
+    role: 'radiogroup' | 'img';
```

Comment:
is this needed? isn't the role determined by `isInteractive` boolean?

Reply:
Yes, I can remove it

Reply:
removed